src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6585,7 +6585,6 @@ public override IOperation VisitInstanceReference(IInstanceReferenceOperation op
                     }
                     else
                     {
```

Comment:
I'd still like to understand how we're getting here, as well as having explicit ioperation tests for it.

Reply:
I think we are getting here because of the cycle - the `_currentImplicitInstance` is unset because the receiver is the same collection expression as the one being visited. I will add tests.

Reply:
I'm not comfortable with getting rid of this assert entirely. Let's assert the specific case we're talking about, so we still catch any other violations.